src/LsmTree.hs

```diff
@@ -10,26 +11,36 @@ module LsmTree
        , emptyTable
        , initTree
        , initTable
+       , insertTree
        ) where
 
 -- | Data type that stores values while program is working.
-data Tree a
+data Ord a => Tree a
     = TreeNode (Tree a) a (Tree a)
     | Leaf
     deriving stock (Eq, Show)
 
 emptyTree :: Tree a
 emptyTree = Leaf
 
-initTree :: a -> Tree a
+initTree :: Ord a => a -> Tree a
 initTree x = TreeNode Leaf x Leaf
 
+insertTree :: Ord a => Tree a -> a -> Tree a
+insertTree Leaf x = initTree x
+insertTree (TreeNode l val r) x
```

Comment:
Traditionally, data structure should be the last argument. To make it easier to insert in chain. With this style you can write:

```haskell
insertTree x = \case
    Leaf -> ...
    TreeNode ... ->
```